allocations/allocation_mapper.py

```diff
@@ -11,45 +34,129 @@ def _wants_db_allocations(self, request):
             return True
         return False
     
-    def map(self, project):
+    def map(self, project, get_redis_data = True):
         if self.allocations_from_db:
+            is_dict_type = isinstance(project, dict)
             balance_service = BalanceServiceClient()
             reformated_allocations = []
-            for alloc in portal_alloc.objects.filter(project_charge_code=project.chargeCode):
-                if alloc.status == 'active':
-                    active_allocation = balance_service.call(project.chargeCode)
+            project_charge_code = project['chargeCode'] if is_dict_type else project.chargeCode
```

Comment:
Another way of doing this:

```python
def get_attr(obj, key):
    """Attempt to resolve the key either as an attribute or a dict key"""
    if isinstance(obj, dict):
        return obj.get(key)
    else:
        return getattr(obj, key, None)

# Later...
project_charge_code = get_attr(project, 'chargeCode')
```

You can then use that instead of having the `is_dict_type` stuff peppered around here.